File: src/main/java/com/pengelkes/controller/TeamController.java
package com.pengelkes.controller;

import com.pengelkes.service.team.Team;
import com.pengelkes.service.team.TeamService;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.web.bind.annotation.RequestBody;
import org.springframework.web.bind.annotation.RequestMapping;
import org.springframework.web.bind.annotation.RequestMethod;
import org.springframework.web.bind.annotation.RestController;

import javax.servlet.ServletException;
import java.util.List;
import java.util.Optional;

/**
 * Created by pengelkes on 02.12.2016.
 */
@RestController
@RequestMapping("/api")
public class TeamController
{
    private TeamService teamService;

    @Autowired
    public TeamController(TeamService teamService)
    {
        this.teamService = teamService;
    }

    @RequestMapping(value = "/teams", method = RequestMethod.GET)
    public List<Team> getAllTeams()
    {
        return teamService.getAllTeams();
    }

    @RequestMapping(value = "/teams", method = RequestMethod.POST)
    public Optional<Team> createTeam(@RequestBody String json) throws ServletException
    {
        Optional<Team> teamOptional = Team.fromJson(json);
        if (teamOptional.isPresent())
        {
            teamOptional = teamService.create(teamOptional.get());
        }

        return teamOptional;
    }
}

File: src/main/java/com/pengelkes/controller/TeamController.kt
package com.pengelkes.controller

import com.pengelkes.service.Team
import com.pengelkes.service.TeamService
import com.pengelkes.service.User
import org.springframework.beans.factory.annotation.Autowired
import org.springframework.web.bind.annotation.*
import javax.servlet.ServletException

/**
 * Created by pengelkes on 02.12.2016.
 */
@RestController
@RequestMapping("/api")
class TeamController
@Autowired
constructor(private val teamService: TeamService) {

    @RequestMapping(value = "/teams", method = arrayOf(RequestMethod.GET))
    fun getAllTeams() = teamService.findAll()

    @RequestMapping(value = "/teams/{id}", method = arrayOf(RequestMethod.GET))
    fun findById(@PathVariable id: Int): Team? {
        return teamService.findById(id);
    }

    @RequestMapping(value = "/teams", method = arrayOf(RequestMethod.POST))
    @Throws(ServletException::class)
    fun add(@RequestBody json: String): Team {
        Team.fromJson(json)?.let {
            return teamService.create(it)
        }

        throw ServletException("Das Team konnte nicht erstellt werden");
    }

    @RequestMapping(value = "/teams/{id}", method = arrayOf(RequestMethod.POST))
    fun update(@RequestBody json: String, @PathVariable id: Int): Team {
        Team.fromJson(json)?.let {
            return teamService.update(it);
        }

        throw ServletException("Das Team konnte nicht erstellt werden");
    }

    @RequestMapping(value = "/teams/{id}/players", method = arrayOf(RequestMethod.GET))
    fun getAllPlayers(@PathVariable id: Int): List<User> {
        return teamService.findAllPlayersByTeam(id)
    }
}
